data/templates/get_driveid.js

```diff
@@ -0,0 +1,196 @@
+        }
+        return [esxiLine[0], strLine];
+    });
+
+    var esxiScsi = scsiLines.map(function(esxiLine) {
+        var line = esxiLine[1];
+        var split = line.split(/-|_/);
+        return [esxiLine[0], 'naa.' + split[1].slice(1)];
+    });
+
+    var linuxParsed = sataLines.concat(scsiLines).map(function(linuxLine) {
+        return [linuxLine[0], '/dev/disk/by-id/' + linuxLine[1]];
+    });
+
+    return {esxiDriveIds: esxiSata.concat(esxiScsi), linuxDriveIds: linuxParsed};
+}
+
+
+/**
+ * Parse Virtual Drive output via by-path output
+ * @param {String} data
+ * @return
+ */
+function parseVdInfo(pathList) {
+    var lines = pathList.split('\n').map(function(line) {
+        var split = line.split(/\s+/);
+        return [split[8],split[10]].join('->');
+    });
+    var pciLines = [];
+    lines.forEach(function(line){
+        if ( line && !(line.match('part'))){
+            var nameIndex = line.lastIndexOf('/'), idIndex = line.lastIndexOf('->');
+            if (line.indexOf('pci') === 0) {
+                pciLines.push([line.slice(nameIndex + 1), line.slice(0,idIndex)]);
+            }
+        }
+    });
+
+    return pciLines.map(function(line) {
+        var scsiId = line[1].slice(line[1].lastIndexOf('-') + 1), vdStr;
+        if (scsiId.split(':').length === 4){
+		    var scsiIdArray = scsiId.split(':');
+            vdStr = ['/c', scsiIdArray[0], '/v', scsiIdArray[2]].join('');
+		}
+	    return [line[0], vdStr, scsiId];
+    });
+}
+
+/**
+* Parse scsi ID output via lsscsi command
+* @param {String} data
+*/
+function parseScsiInfo(lsscsiList) {
+	var lines = lsscsiList.split('\n');
+	return lines.map(function(line) {
+		if(line){
+		return [line.slice(line.lastIndexOf('/') + 1).replace(' ', ''),
+            line.slice(1, line.indexOf(']'))];
+		}
+	});
+}
+
+/**
+ * Build the drive mapping table
+ * @param {String} wwidData
+ * @param {String} vdData
+ * @return 0 if success, otherwise failed.
+ */
+function buildDriveMap(wwidData, vdData, scsiData) {
+    var parsedWwids = parseDriveWwid(wwidData),
+        scsiList = parseScsiInfo(scsiData),
+        vdList = parseVdInfo(vdData);
+    var linuxWwids = parsedWwids.linuxDriveIds, esxiWwids = parsedWwids.esxiDriveIds;
+    var driveIds=[];
+    esxiWwids.forEach(function(esxiWwid){
+        var vd = '', scsiId = '', diskPath = esxiWwid[0];
+        vdList.forEach(function(elem){
+            if (typeof elem !== 'undefined') {
+                if (elem[0] === diskPath) {
+                    vd = elem[1];
+                }
+            }
+        });
+        scsiList.forEach(function(elem){
+            if (typeof elem !== 'undefined'){
+                if(elem[0] === diskPath){
+                    scsiId = elem[1];
+                }
+            }
+        });
+        driveIds.push({"scsiId": scsiId, "virtualDisk": vd,
+            "esxiWwid": esxiWwid[1], "devName": diskPath});
+    });
+    var k=0;
+    linuxWwids.forEach(function(linuxWwid){
```

Comment:
The internal function for "forEach" can be changed to function(linuxWwid, k), so we don't need to create another index variable "k".